vncdotool/client.py

```diff
@@ -336,6 +337,22 @@ def mouseDrag(self, x, y, step=1):
 
         return self
 
+    def setImageMode(self):
+        """ Extracts color ordering and 24 vs. 32 bpp info out of the pixel format information
+        """
+        if (self.truecolor and (not self.bigendian) and self.depth == 24
+                and self.redmax == 255 and self.greenmax == 255 and self.bluemax == 255):
```

Comment:
To check I'm understanding this correctly, this is testing we're using a 3 channel 8bit colour map?

Reply:
Yeah. Basically, it can handle different color orderings, and with or without a padding / alpha byte, but that's it. So any attribute of the pixel format that I don't explicitly use to determine the image mode, I explicitly assert on here.

Reply:
cool, I like it also opens the door to other formats being implemented, as needed. There are failing tests, seems like truecolor isn't being handled consistently.  Once those are fixed I'll merge this.